src/PIL/EpsImagePlugin.py

```diff
@@ -275,43 +317,44 @@ def _open(self):
                     else:
                         msg = "bad EPS header"
                         raise OSError(msg)
-
-            s_raw = fp.readline()
-            s = s_raw.strip("\r\n")
-
-            if s and s[:1] != "%":
-                break
-
-        #
-        # Scan for an "ImageData" descriptor
-
-        while s[:1] == "%":
-            if len(s) > 255:
-                msg = "not an EPS file"
-                raise SyntaxError(msg)
-
-            if s[:11] == "%ImageData:":
-                # Encoded bitmapped image.
-                x, y, bi, mo = s[11:].split(None, 7)[:4]
-
-                if int(bi) == 1:
+            elif bytes_mv[:11] == b"%ImageData:":
+                # Check for an "ImageData" descriptor
+                # https://www.adobe.com/devnet-apps/photoshop/fileformatashtml/#50577413_pgfId-1035096
+
+                # Values:
+                # columns
+                # rows
+                # bit depth (1 or 8)
+                # mode (1: L, 2: LAB, 3: RGB, 4: CMYK)
+                # number of padding channels
+                # block size (number of bytes per row per channel)
+                # binary/ascii (1: binary, 2: ascii)
+                # data start identifier (the image data follows after a single line
+                #   consisting only of this quoted value)
+                image_data_values = byte_arr[11:bytes_read].split(None, 7)
+                columns, rows, bit_depth, mode_id = [
+                    int(value) for value in image_data_values[:4]
+                ]
+
+                if bit_depth == 1:
                     self.mode = "1"
-                elif int(bi) == 8:
+                elif bit_depth == 8:
                     try:
-                        self.mode = self.mode_map[int(mo)]
+                        self.mode = self.mode_map[mode_id]
                     except ValueError:
                         break
                 else:
                     break
 
-                self._size = int(x), int(y)
+                self._size = columns, rows
                 return
 
-            s = fp.readline().strip("\r\n")
-            if not s:
-                break
+            bytes_read = 0
+
+        check_required_header_comments()
 
-        if not box:
+        if not self._size:
+            self._size = 1, 1  # errors if this isn't set. why (1,1)?
```

Comment:
Could you tell me what the "errors" are? I don't see any coming from our test suite - https://github.com/radarhere/Pillow/actions/runs/4582259664

Reply:
I've created PR #7056 to remove this line.

Reply:
I checked out the first commit in this branch, removed that line, and ran the tests, and there weren't any errors. So I'm not sure what error I had seen back then.